sync/src/relayer/compact_block_process.rs

```diff
@@ -260,25 +260,21 @@ impl<'a> CompactBlockProcess<'a> {
         } else {
             StatusCode::CompactBlockRequiresFreshTransactions.with_context(&block_hash)
         };
-        if !missing_transactions.is_empty() {
-            metrics!(value, "ckb-net.fresh", missing_transactions.len() as u64, "type" => "transactions", "status" => status.tag());
-        }
-        if !missing_uncles.is_empty() {
-            metrics!(value, "ckb-net.fresh", missing_uncles.len() as u64, "type" => "uncles", "status" => status.tag());
-        }
 
         let content = packed::GetBlockTransactions::new_builder()
             .block_hash(block_hash)
             .indexes(missing_transactions.pack())
             .uncle_indexes(missing_uncles.pack())
             .build();
         let message = packed::RelayMessage::new_builder().set(content).build();
-
-        if let Err(err) = self.nc.send_message_to(self.peer, message.as_bytes()) {
-            return StatusCode::Network
-                .with_context(format!("Send GetBlockTransactions error: {:?}", err));
+        let sending = send_message_to(self.nc.as_ref(), self.peer, &message);
+        if !sending.is_ok() {
```

Comment:
can we remove this redundant log here since we have log the sending error in `send_message` fn already?

Reply:
You are right that `fn send_message` logs when an error occurs. While I think that error log cannot provide us the context, where the sending behavior comes from, whether I should ignore it?
Therefore I leave warning logs so that users can get more context when an error occurs.
Yeah, it's a bit redundant, but it's worthy.